diff --git a/password_policy.install b/password_policy.install
index d761ccfa640b1451f20db02cddf0219882356039..eb9e2876f5faba5f94813ecfeb030ac094174e53 100644
--- a/password_policy.install
+++ b/password_policy.install
@@ -12,46 +12,48 @@ use Drupal\datetime\Plugin\Field\FieldType\DateTimeItemInterface;
 /**
  * Implements hook_install().
  */
-function password_policy_install() {
-  // Give precedence to modules like prlp or simple_pass_reset.
-  module_set_weight('password_policy', 10);
 
-  // Set user password reset timestamp to now.
-  $timestamp = \Drupal::service("date.formatter")->format(\Drupal::time()->getRequestTime(), "custom", DateTimeItemInterface::DATETIME_STORAGE_FORMAT, DateTimeItemInterface::STORAGE_TIMEZONE);
-  \Drupal::state()->set('password_policy.install_time', $timestamp);
-
-  // Rebuild user entity form display for new fields.
-  $storage = \Drupal::entityTypeManager()->getStorage('entity_form_display');
-  /** @var \Drupal\Core\Entity\Display\EntityFormDisplayInterface $user_form_display */
-  $user_form_display = $storage->load('user.user.default');
-  if (!$user_form_display) {
-    $user_form_display = $storage->create([
-      'targetEntityType' => 'user',
-      'bundle' => 'user',
-      'mode' => 'default',
-      'status' => TRUE,
-    ]);
+function password_policy_install($is_syncing) {
+  if (!$is_syncing) {
+    // Set user password reset timestamp to now.
+    $timestamp = \Drupal::service("date.formatter")->format(\Drupal::time()->getRequestTime(), "custom", DateTimeItemInterface::DATETIME_STORAGE_FORMAT, DateTimeItemInterface::STORAGE_TIMEZONE);
+    \Drupal::state()->set('password_policy.install_time', $timestamp);
+
+    // Rebuild user entity form display for new fields.
+    $storage = \Drupal::entityTypeManager()->getStorage('entity_form_display');
+    /** @var \Drupal\Core\Entity\Display\EntityFormDisplayInterface $user_form_display */
+    $user_form_display = $storage->load('user.user.default');
+    if (!$user_form_display) {
+      $user_form_display = $storage->create([
+        'targetEntityType' => 'user',
+        'bundle' => 'user',
+        'mode' => 'default',
+        'status' => TRUE,
+      ]);
+    }
+    $user_form_display
+      ->setComponent('field_last_password_reset', [
+        'type' => 'datetime_default',
+        // Display before contact standard profile field.
+        'weight' => 4,
+        'settings' => [],
+      ])
+      ->setComponent('field_password_expiration', [
+        'type' => 'boolean_checkbox',
+        'weight' => 3,
+        'settings' => ['display_label' => TRUE],
+      ])
+      ->save();
   }
-  $user_form_display
-    ->setComponent('field_last_password_reset', [
-      'type' => 'datetime_default',
-      // Display before contact standard profile field.
-      'weight' => 4,
-      'settings' => [],
-    ])
-    ->setComponent('field_password_expiration', [
-      'type' => 'boolean_checkbox',
-      'weight' => 3,
-      'settings' => ['display_label' => TRUE],
-    ])
-    ->save();
 }
 
 /**
  * Implements hook_uninstall().
  */
-function password_policy_uninstall() {
-  \Drupal::state()->delete('password_policy.install_time');
+function password_policy_uninstall($is_syncing) {
+  if (!$is_syncing){
+    \Drupal::state()->delete('password_policy.install_time');
+  }
 }
 
 /**
diff --git a/password_policy_history/password_policy_history.install b/password_policy_history/password_policy_history.install
index 1c8579fb7caa5e6c52f827d49ede012f42ff29bc..81e10074ede05be76fd11156561fb8a05a9e874b 100644
--- a/password_policy_history/password_policy_history.install
+++ b/password_policy_history/password_policy_history.install
@@ -11,22 +11,24 @@ use Drupal\user\Entity\User;
 /**
  * Implements hook_install().
  */
-function password_policy_history_install() {
-  // Add current user passwords.
-  $users = User::loadMultiple();
-  $connection = Database::getConnection();
+function password_policy_history_install($is_syncing) {
+  if (!$is_syncing){
+    // Add current user passwords.
+    $users = User::loadMultiple();
+    $connection = Database::getConnection();
 
-  foreach ($users as $user) {
-    $hashed_pass = $user->getPassword();
-    if ($hashed_pass) {
-      $values = [
-        $user->id(),
-        $hashed_pass,
-        \Drupal::time()->getRequestTime(),
-      ];
-      $connection->insert('password_policy_history')
-        ->fields(['uid', 'pass_hash', 'timestamp'], $values)
-        ->execute();
+    foreach ($users as $user) {
+      $hashed_pass = $user->getPassword();
+      if ($hashed_pass) {
+        $values = [
+          $user->id(),
+          $hashed_pass,
+          \Drupal::time()->getRequestTime(),
+        ];
+        $connection->insert('password_policy_history')
+          ->fields(['uid', 'pass_hash', 'timestamp'], $values)
+          ->execute();
+      }
     }
   }
 }
